### ISSUE TITLE
Latest less.js (2.1.1) behaves async in IE

### ISSUE THREAD

**Issue by lukeapage** [less/less.js#2317]
Latest less.js (2.1.1) behaves async in IE (ver 11) even when async:false is set explicitly.
The file processed is complex less with multiple @import commands that take about 1 second to load and process on dev machine, during which time IE shows style-free html. Firefox and Chrome behave correctly. Had to revert to 1.7.5 which behaves correctly.
Perhaps related to the recent promises changes? Perhaps setTimeout(function(){}, 0); allows IE to proceed in parallel and partially render the page?

**Comment by efbiaiinzinz** [comment]
Latest less.js (2.3.1) still works quirky in the Firefox. Works correctly in Chrome and surprisingly also works correctly in IE11.
Examples of page loading in FF:
Plain html without ay style:
![image](https://cloud.githubusercontent.com/assets/5718209/6056565/485deb56-ad1b-11e4-97be-b2ffba8b6c89.png)
Partial style:
![image](https://cloud.githubusercontent.com/assets/5718209/6056575/72f7006e-ad1b-11e4-8119-3fb9795a9b09.png)
Final rendering:
![image](https://cloud.githubusercontent.com/assets/5718209/6056585/95d6e784-ad1b-11e4-9c89-e868f453e95c.png)

In IE and Chrome, the page stays blank until less is processed and then does the final rendering at once.

**Comment by chipx86** [comment]
We've hit this problem as well, and can verify this behavior on Safari and Firefox (haven't tried IE yet). Chrome works fine. I've tried every version from 2.0 beta 1 on through 2.5, and this problem exists in all of them.

So, I bisected the tree, and this appears to have been introduced in d31be578a3b35c7b0f43738e26f42576bab48ce9.

This problem is more than just annoying. It's actually caused some very bad interactions with some JavaScript on the page, which expected CSS to be loaded properly at the time of execution. Unfortunately, that means we're stuck on 1.7.5 for the time-being (not a huge deal, though we are hitting some bugs with it that have been fixed since..).

I spent some time investigating this change, the referenced bug numbers, and the pull request introducing it. The intent, it seems, was to add a function (`less.registerStylesheets`) that other callers can call that can return a `Promise` and then allow the caller to call `less.refresh` or whatever it wants. That seems like a great addition, but it's unclear whether it was intended to change the initial load behavior.

I've changed this locally (just to test) to move the bulk of `less.registerStylesheets` out into a `less.registerStylesheetsImmediately`, and to have `less.registerStylesheets` wrap that in a `Promise`. Then, the initial page load calls `less.registerStylesheetsImmediately`, followed by `less.refresh`, without using promises. This returns to the original loading behavior from 1.7.5 in Firefox. I'm happy to supply a pull request for that, if the general idea sounds good.

Unfortunately, that didn't solve the issue entirely for Safari (though it definitely improved it). More bisecting led me to commit 544bd3abf084f7ab5586d7b578d3e37494cebc31. It's 4:15AM though, and I'm falling asleep, so that's as far as I'm getting tonight.

Hopefully that helps someone in diagnosing the cause. Would love to see this fixed :)

**Comment by chipx86** [comment]
Following up on this.

The Safari-specific issue I mentioned was noticed when I had tested my fix on top of commit d31be57. Having now ported my patch up to 2.5.0, I realize that the Safari issue had already been fixed in some prior commit, which is great.

I'm working on a patch for the above solution, assuming it sounds fine.

**Comment by chipx86** [comment]
Just a note on the change I have up for review: There's a bug it uncovered in GitHub's diff viewer, causing a line to disappear from the change. Right before this line:

```javascript
less.pageLoadFinished = less.refresh(less.env === 'development');
```

There's:

```javascript
less.registerStylesheetsImmediately();
```

Feel free to verify from my tree. Already reporting it to GitHub.
